src/core/server/integration_tests/saved_objects/migrations/group4/v2_with_mv_same_stack_version.test.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import fs from 'fs/promises';
+import { range, sortBy } from 'lodash';
+import { type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import { SavedObjectsBulkCreateObject } from '@kbn/core-saved-objects-api-server';
+import { modelVersionToVirtualVersion } from '@kbn/core-saved-objects-base-server-internal';
+import '../jest_matchers';
+import { getKibanaMigratorTestKit, startElasticsearch } from '../kibana_migrator_test_kit';
+import { delay, createType, parseLogFile } from '../test_utils';
+import { getBaseMigratorParams } from '../fixtures/zdt_base.fixtures';
+
+const logFilePath = Path.join(__dirname, 'v2_with_mv_same_stack_version.test.log');
+
+const NB_DOCS_PER_TYPE = 25;
+
+describe('V2 algorithm - using model versions - upgrade without stack version increase', () => {
```

Comment:
Standard upgrade scenario where the stack version doesn't change (only the model versions of the registered types change)